### ISSUE TITLE
List View: Cannot collapse parent Block which contains child Blocks with screen reader

### ISSUE THREAD

**Issue by alexstine** [WordPress/gutenberg#37972]
### Description

The List View does not allow you to collapse a parent Block such as Columns Block while using a screen reader.

### Step-by-step reproduction instructions

1. Open a Post or Page for editing.
2. Insert some Blocks.
3. Insert a Columns Block and choose a variation from the list.
4. Open the List View.
5. With the keyboard, navigate to the Columns Block.
6. Notice how pressing Left Arrow on the Columns Block does not collapse the children Column Blocks within.

### Environment info

Browser: Firefox
Screen Reader: NVDA
OS: Windows 10 Professional
Gutenberg Version: trunk

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by talldan** [comment]
I gave it a test on Mac OS X (Voiceover and Safari) and it worked ok. Unfortunately I don't have a way to test Windows since switching to an M1 Mac.

**Comment by alexstine** [comment]
@talldan It's the keycode detection messing this up on Windows. I'm still looking in to it.

**Comment by alexstine** [comment]
Part of me wonders if it is useCallback causing some kind of issue. wordpress/keycodes is used plenty of other places and not causing issues. It's like useCallback() is only allowing certain keys to execute instead of every keypress.

**Comment by talldan** [comment]
`useCallback` isn't something that I'd expect to work differently between browsers, so it's definitely an unusual one.

Looking at it now, the dependencies array of `useCallback` looks like it's missing a couple of things. `onExpandRow` and `onCollapseRow` should be specified because they can't be guaranteed as stable references.

**Comment by tellthemachines** [comment]
I can reproduce this on Firefox on both Windows 10 and Mac OS X, but not on any other browser (Chrome on Windows works fine, with or without NVDA). So I'm guessing it's a Firefox-specific issue.
